asttokens/asttokens.py

```diff
@@ -66,56 +66,65 @@ def __init__(self, source_text, parse=False, tree=None, filename='<unknown>'):
 
 
   def mark_tokens(self, root_node):
+    # type: (Module) -> None
     """
     Given the root of the AST or Astroid tree produced from source_text, visits all nodes marking
     them with token and position information by adding ``.first_token`` and
     ``.last_token``attributes. This is done automatically in the constructor when ``parse`` or
     ``tree`` arguments are set, but may be used manually with a separate AST or Astroid tree.
     """
     # The hard work of this class is done by MarkTokens
+    from .mark_tokens import MarkTokens # to avoid import loops
     MarkTokens(self).visit_tree(root_node)
 
 
   def _generate_tokens(self, text):
+    # type: (str) -> Iterator[Token]
     """
     Generates tokens for the given code.
     """
     # This is technically an undocumented API for Python3, but allows us to use the same API as for
     # Python2. See http://stackoverflow.com/a/4952291/328565.
-    for index, tok in enumerate(tokenize.generate_tokens(io.StringIO(text).readline)):
+    for index, tok in enumerate(tokenize.generate_tokens(six.StringIO(text).readline)):
```

Comment:
Since you're having Python 2 unicode issues, can you revert this and see if it helps?

Reply:
Resolved. [Upstream typing issue](https://github.com/python/typeshed/issues/7003) and there's now a `cast` to fix the type as a workaround.